src/openms/source/FORMAT/PercolatorInfile.cpp

```diff
@@ -157,6 +199,11 @@ namespace OpenMS
         }
       }
 
+      if (charge != 0)
+      {
+        pids.back().setMZ((row[to_idx.at("ExpMass")].toDouble() - std::fabs(charge) * Constants::PROTON_MASS_U) / std::fabs(charge));
```

Comment:
not sure I understand this formula.
What is "ExpMass" ?
If it's the uncharged mass, then protons should be added to obtain m/z, not subtracted?!

Reply:
it is the charged mass

Reply:
digging a bit deeper it seems that percolator doesn't really care. It just needs both ExpMass and CalcMass calculated the same way. Some software output m/z some M some [M+H]z+. Here I used the sage output as reference